fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/rules/RemoveSortRule.java

```diff
@@ -74,12 +73,7 @@ public void onMatch(@Nonnull PlannerRuleCall call) {
             return;
         }
 
-        final Optional<Ordering> orderingOptional = OrderingProperty.evaluate(innerPlan, call.getContext());
-        if (orderingOptional.isEmpty()) {
-            return;
-        }
-
-        final Ordering ordering = orderingOptional.get();
+        final Ordering ordering = OrderingProperty.OrderingVisitor.evaluate(innerPlan);
```

Comment:
Note to reviewer: This call uses the memoized ordering information. The reason we this rule still matches on `anyPlan()` instead of on a plan partition is the nature of this rule to just eliminate expressions.